vector/src/main/java/im/vector/app/core/time/Clock.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.time
+
+import javax.inject.Inject
+
+interface Clock {
+    fun epochMillis(): Long
```

Comment:
It's worth noting that the current impl rely on the system date, which can be set by the user (and at any time).
This can lead to bugs...
Also when the clock change automatically (Winter?Summer time) we can have bugs (my experience is talking here).

Reply:
unless we use a server time is there a way to avoid this? analytics SDKs will rely on a mix of server _received at_ timestamps and client _sent at_

the `currentTimeMillis` epoch is UTC, unless a user sets a custom time using their local time (eg UTC+11) but with UTC+0 timezone :sweat_smile:   

it's becoming rarer for users to misconfigure their time due to automatic network times and a lot apps/websites start complaining about SSL errors if the current time/date is wildly wrong

Reply:
https://github.com/vector-im/element-android/pull/4556/commits/fdfac8d20acad014090c90413476e3d7328ce1c7